crypto/asn1/a_object.c

```diff
@@ -184,15 +184,15 @@ ASN1_OBJECT *c2i_ASN1_OBJECT(ASN1_OBJECT **out, const unsigned char **inp,
 ASN1_OBJECT *ASN1_OBJECT_new(void) {
   ASN1_OBJECT *ret;
 
-  ret = (ASN1_OBJECT *)OPENSSL_malloc(sizeof(ASN1_OBJECT));
+  ret = (ASN1_OBJECT *)OPENSSL_zalloc(sizeof(ASN1_OBJECT));
   if (ret == NULL) {
     return NULL;
   }
-  ret->length = 0;
-  ret->data = NULL;
-  ret->nid = 0;
-  ret->sn = NULL;
-  ret->ln = NULL;
+  assert(ret->length == 0);
+  assert(ret->data == NULL);
+  assert(ret->nid == 0);
+  assert(ret->sn == NULL);
+  assert(ret->ln == NULL);
```

Comment:
We don't consistently `assert` on every structure in this pull request, is this providing us any benefit? I feel like a test-case to demonstrate `OPENSSL_zalloc` behavior would be more appropriate to verify it calls calloc. Is more appropriate? Is `OPENSSL_zalloc` allowed to be overridden with a user provided implementation similar to malloc, is that concern?

Reply:
> We don't consistently assert on every structure in this pull request, is this providing us any benefit?

I converted initializations that were now redundant into (debug) `assert` statements. The benefit is arguably minor: (*) documenting the (now implicit) initial value of these fields, and (*) verifying that my change is preserving its initialized value.

Reply:
In the [latest commit](https://github.com/aws/aws-lc/pull/1447/commits/396ac54acf46209f3dbfb6b42096e61c616ef734), I removed many of these debug asserts.